extensions/ql-vscode/src/remote-queries/sarif-processor.ts

```diff
@@ -0,0 +1,137 @@
+import * as sarif from 'sarif';
+
+import { AnalysisAlert, ResultSeverity } from './shared/analysis-result';
+
+export class ProcessedSarif {
+  public constructor(
+    public readonly alerts: AnalysisAlert[],
+    public readonly errors: string[]) { }
+}
+
+export function processSarif(sarifLog: sarif.Log): ProcessedSarif {
```

Comment:
Could you add some unit tests for this function. It is nicely contained and should be straight forward.

Reply:
Thanks for the nudge, I was being lazy. I've added a few unit tests for the various functions. They don't cover everything as there are way too many possibilities, but I think they cover enough.